lib/salus/scanners/osv/maven_osv.rb

```diff
@@ -0,0 +1,150 @@
+require 'salus/scanners/osv/base'
+
+module Salus::Scanners::OSV
+  class MavenOSV < Base
+    class SemVersion < Gem::Version; end
+
+    EMPTY_STRING = "Not Found".freeze
+    DEFAULT_SOURCE = "https://osv.dev/list".freeze
+    DEFAULT_SEVERITY = "MODERATE".freeze
+    GITHUB_DATABASE_STRING = "Github Advisory Database".freeze
+    MAVEN_OSV_ADVISORY_URL = "https://osv-vulnerabilities.storage.googleapis.com/"\
+        "Maven/all.zip".freeze
+
+    def should_run?
+      @repository.pom_xml_present?
+    end
+
+    def run
+      dependencies = find_dependencies
+      if dependencies.empty?
+        err_msg = "MavenOSV: Failed to parse any dependencies from the project."
+        report_stderr(err_msg)
+        report_error(err_msg)
+        return
+      end
+
+      @osv_vulnerabilities ||= fetch_vulnerabilities(MAVEN_OSV_ADVISORY_URL)
+      if @osv_vulnerabilities.nil?
+        err_msg = "No vulnerabilities found to compare."
+        bugsnag_notify("MavenOSV: #{err_msg}")
+        return report_error("MavenOSV: #{err_msg}")
+      end
+
+      # Fetch vulnerable dependencies.
+      # Dedupe and select Github Advisory over other sources if available.
+      results = []
+      grouped = match_vulnerable_dependencies(dependencies).group_by { |d| d[:ID] }
+      grouped.each do |_key, values|
+        vuln = {}
+        values.each do |v|
+          vuln = v if v[:Database] == GITHUB_DATABASE_STRING
+        end
+        results.append(vuln.empty? ? values[0] : vuln)
+      end
+
+      # Report scanner status
+      return report_success if results.empty?
+
+      report_failure
+      log(JSON.pretty_generate(results))
+    end
+
+    # Match if dependency version found is in the range of
+    # vulnerable dependency found.
+    def version_matching(version, version_ranges)
+      vulnerable_flag = false
+      version_found = SemVersion.new(version)
+      # If version range length is 1, then no fix available.
+      if version_ranges.length == 1
+        introduced = SemVersion.new(
+          version_ranges[0]["introduced"]
```

Comment:
if length is 1, could it the `introduced` key still not exist?

Reply:
Same response as above - 
Could be (for example - if the input we get is bad) but I haven't seen an example of it since they are following the OSV Format. So according the format these fields would need to be there.